src/acrcssc/azext_acrcssc/helper/_workflow_status.py

```diff
@@ -0,0 +1,600 @@
+                 f"\tscan task ID: {status.scan_task_id}\n"
+
+        if hasattr(status, "scan_error_reason"):
+            result += f"\tscan error reason: {status.scan_error_reason}\n"
+
+        result += f"\tpatch status: {status.patch_status}\n"
+
+        if hasattr(status, "patch_error_reason"):
+            result += f"\tpatch error reason: {status.patch_error_reason}\n"
+
+        if hasattr(status, "patch_skipped_reason"):
+            result += f"\tpatch skipped reason: {status.patch_skipped_reason}\n"
+
+        result += f"\tpatch date: {status.patch_date}\n" \
+                  f"\tpatch task ID: {status.patch_task_id}\n" \
+                  f"\tlast patched image: {status.patched_image}\n" \
+                  f"\tworkflow type: {status.workflow_type}"
+
+        return result
+
+    @staticmethod
+    def generate_logs(cmd,
+                      client,
+                      run_id,
+                      registry_name,
+                      resource_group_name,
+                      await_task_run=True,
+                      await_task_message=None):
+
+        log_file_sas = None
+        error_msg = "Could not get logs for ID: {}".format(run_id)
+        try:
+            response = client.get_log_sas_url(
+                resource_group_name=resource_group_name,
+                registry_name=registry_name,
+                run_id=run_id)
+            log_file_sas = response.log_link
+        except (AttributeError, CloudError) as e:
+            logger.debug("%s Exception: %s", error_msg, e)
+            raise AzCLIError(error_msg)
+        except ResourceNotFoundError as e:
+            logger.debug("log file not found for run_id: %s, registry: %s, "
+                         "resource_group: %s -- exception: %s",
+                         run_id, registry_name, resource_group_name, e)
+
+        if await_task_run:
+            try:
+                polling_method = WorkflowLogPollingMethod(client,
+                                                          resource_group_name,
+                                                          registry_name,
+                                                          run_id)
+                result = LongRunningOperation(
+                    cmd.cli_ctx,
+                    progress_bar=IndeterminateProgressBar(cmd.cli_ctx, message=await_task_message),
+                    poller_done_interval_ms=1500  # every poller call will do a call to the API
+                )(polling_method)
+                logger.debug("Task result: %s", result)
+            except TimeoutError:
+                logger.debug("Timeout waiting for task run to complete, workflow task run ID: %s", run_id)
+                logger.debug("An attempt to retrieve the logs will be done, if there are any")
+
+        blobClient = get_sdk(cmd.cli_ctx, ResourceType.DATA_STORAGE_BLOB, '_blob_client#BlobClient')
+        return WorkflowTaskStatus._download_logs(blobClient.from_blob_url(log_file_sas))
```

Comment:
Do you need add try block here to catch the exception thrown during download logs?

Reply:
added to `_download_logs()` so it returns an empty string in case of exception and log the issue